### PR TITLE
fix crash of backgrouding in iOS16

### LINKED ISSUE #801: Version 7.6.5 will CRASH on iOS 16 due to removed `kCFStreamNetworkServiceTypeVoIP`
As [reported](https://github.com/robbiehanson/CocoaAsyncSocket/issues/361#issuecomment-1227076872) the latest version will crash on iOS 16, since Apple finally removed constant `kCFStreamNetworkServiceTypeVoIP` which is still be used by AsyncSocket.

I have provided a solution for issue https://github.com/robbiehanson/CocoaAsyncSocket/issues/361 more than 2 years ago via PR #717 but this has never be implemented.
Please release an update asap where `kCFStreamNetworkServiceTypeVoIP` should be replaced by `kCFStreamNetworkServiceTypeBackground` as indicated in the above mentioned comments and PR.

### PR DESCRIPTION
**Pull request by JonyFang**
fix #801

### CODE REVIEW

**Review by michalsrutek — APPROVED** [review]

**Comment by michalsrutek** [comment]
@robbiehanson can we get this merged and released? 🙏

**Comment by leeway1208** [comment]
> @robbiehanson can we get this merged and released? 🙏

@chrisballinger

**Comment by NicoDB79** [comment]
I have the same issue using this pod as a dependency of the CocoaMQTT library. Can you please merge and release?
Thanks

File: Source/GCD/GCDAsyncSocket.m [hunk 1]
```diff
@@ -8200,8 +8200,8 @@ - (BOOL)enableBackgroundingOnSocketWithCaveat:(BOOL)caveat
 	
 #pragma clang diagnostic push
 #pragma clang diagnostic ignored "-Wdeprecated-declarations"
-	r1 = CFReadStreamSetProperty(readStream, kCFStreamNetworkServiceType, kCFStreamNetworkServiceTypeVoIP);
-	r2 = CFWriteStreamSetProperty(writeStream, kCFStreamNetworkServiceType, kCFStreamNetworkServiceTypeVoIP);
+	r1 = CFReadStreamSetProperty(readStream, kCFStreamNetworkServiceType, kCFStreamNetworkServiceTypeBackground);
+	r2 = CFWriteStreamSetProperty(writeStream, kCFStreamNetworkServiceType, kCFStreamNetworkServiceTypeBackground);
```
Comment (mh-ehedstrom) [inline Source/GCD/GCDAsyncSocket.m:8204]: We should update all the examples in the comments too, right?

Comment (craigt543nz) [reply Source/GCD/GCDAsyncSocket.m:8204]: I have the same issue. Thanks for the interim fix and looking forward the merge and release.

Comment (JonyFang) [reply Source/GCD/GCDAsyncSocket.m:8204]: > We should update all the examples in the comments too, right?

yep, in the new commit, it has been updated.

**Comment by Massmaker** [comment]
I made locally the same changes to the source code file. It helps. the app is not crashing.

**Comment by dauerr** [comment]
I forked the project and applied the fix. Works with iOS 16.0 and 16.1. If anyone wants to use it add this to your podfile until this PR is merged:

pod 'CocoaAsyncSocket', :git => 'https://github.com/dauerr/CocoaAsyncSocket'
pod 'CocoaMQTT', :git => 'https://github.com/dauerr/CocoaMQTT'

**Comment by blusewang** [comment]
> 

Thanks You saved me！

**Comment by blusewang** [comment]
> I forked the project and applied the fix. Works with iOS 16.0 and 16.1. If anyone wants to use it add this to your podfile until this PR is merged:
> 
> pod 'CocoaAsyncSocket', :git => 'https://github.com/dauerr/CocoaAsyncSocket' pod 'CocoaMQTT', :git => 'https://github.com/dauerr/CocoaMQTT'

Thanks You saved me！

File: Source/GCD/GCDAsyncSocket.m [hunk 2]
```diff
@@ -8200,8 +8200,8 @@ - (BOOL)enableBackgroundingOnSocketWithCaveat:(BOOL)caveat
 	
 #pragma clang diagnostic push
 #pragma clang diagnostic ignored "-Wdeprecated-declarations"
```
Comment (ApolloZhu) [inline Source/GCD/GCDAsyncSocket.m:8202]: Since there are no more deprecation warnings, I assume we can remove these (and the one below) now?

**Comment by funnel20** [comment]
I'm not sure why there is a big green checkmark saying **This branch has no conflicts with the base branch**. 
There are conflicts, since this PR seems to solve the same issue as PR #804 but in a different way. PR #804 uses `PushKit`, but only solves the issue partially, since there are more occurrences of `kCFStreamNetworkServiceTypeVoIP` throughout the project, as indicated by the changes in this PR (#802).

Both PR #802 and #804 have changed lines 8203 and 8204 of GCDAsyncSocket.m. **So there is a conflict.**

**Comment by mh-ehedstrom** [comment]
> Both PR #802 and #804 have changed lines 8203 and 8204 of GCDAsyncSocket.m. **So there is a conflict.**

Neither PR has been merged into the robbiehanson:master branch, so neither PR has a conflict with the base branch currently. It doesn't say there is no conflict with any other pull requests. As soon as one PR is merged the other will have a conflict.

**Comment by harshadcpatel** [comment]
@robbiehanson can you please merge this PR and release new tag?

**Pull request closed without merging**